packages/shared_preferences/shared_preferences/lib/tools/legacy_to_async_migration_tool.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:shared_preferences_platform_interface/types.dart';
+
+import '../shared_preferences.dart';
+
+/// A tool to migrate from the legacy SharedPreferences system to
+/// SharedPreferencesAsync.
+///
+/// [legacySharedPreferencesInstance] should be an instance of [SharedPreferences]
+/// that has been instantiated the same way it has been used throughout your app.
+/// If you have called [SharedPreferences.setPrefix] that must be done before using
+/// this tool.
+///
+/// [sharedPreferencesAsyncOptions] should be an instance of [SharedPreferencesOptions]
+/// that is set up the way you intend to use the new system going forward.
+/// This tool will allow for future use of [SharedPreferencesAsync] and [SharedPreferencesWithCache].
+///
+/// The [migrationCompletedKey] is a key that will be used to check if the migration
+/// has run before, to avoid overwriting new data going forward. Make sure that
+/// there will not be any collisions with preferences you are or will be setting
+/// going forward, or there may be data loss.
+Future<void> migrateLegacySharedPreferencesToSharedPreferencesAsync(
+  SharedPreferences legacySharedPreferencesInstance,
+  SharedPreferencesOptions sharedPreferencesAsyncOptions,
+  String migrationCompletedKey, {
+  bool clearLegacyPreferences = false,
```

Comment:
This appears to not do anything?

Reply:
forgot about that. Is this a feature that's worth adding? I hadn't yet as it seemed like it could be problematic.

Reply:
We could wait and see if anyone asks. On one hand, I would generally expect a migration to move data instead of duplicating it. On the other hand, it shouldn't be a lot of data.

My one real hesitation here is that if someone isn't using an allow list, on most platforms this would mean that an unprefixed new-style prefs object would be loading all the old prefs too.

Reply:
It also means that items not set by our plugin would get deleted. Potentially if not used carefully (or written carefully) it could write to the same file, then delete the prefs it just wrote.